lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -1280,12 +1280,15 @@ static void lookupVisibleDeclsImpl(VisibleDeclConsumer &Consumer,
       dcGenericParams = dcGenericParams->getOuterParameters();
     }
 
-    if (ExtendedType)
-      ::lookupVisibleMemberDecls(ExtendedType, Consumer, DC, LS, Reason,
+    if (ExtendedType) {
+      ::lookupVisibleMemberDecls(ExtendedType, Consumer, DC, LS, MemberReason,
                                  nullptr);
 
+      // Going outside the current type context.
+      MemberReason = DeclVisibilityKind::MemberOfOutsideNominal;
+    }
+
     DC = DC->getParent();
-    Reason = DeclVisibilityKind::MemberOfOutsideNominal;
```

Comment:
This is a bug fix for:
```
struct S {
  func foo() {}
  func test() {
    _ = { #^COMPLETE^# }
  }
}
```
`foo()` used to be reported as `MemberOfOutsideNominal`, but it actually should be `MemberOfCurrentNominal`